Sanitizer

kalilistic opened the pull request:
Sanitizer from discussion in discord. Intended for cleaning up strings to use with imgui to avoid ???. Accepts individual or lists of strings and language overrides.

NotAdam commented on Dalamud/Plugin/Sanitizer/Sanitizer.cs:
```diff
@@ -0,0 +1,113 @@
+    /// </summary>
+    public class Sanitizer : ISanitizer
+    {
+        private readonly KeyValuePair<string, string> softHyphen = new KeyValuePair<string, string>("\u0002\u0016\u0001\u0003", string.Empty);
```
why? just put it in a dict by language

NotAdam commented:
also whats the point of tests that don't test anything that it actually does

kalilistic commented:
Will do a new PR later with changes.

kalilistic commented:
For ackwell.

kalilistic commented:
- Replaced most hard-coded characters with 0x20 filter
- Replaced kvp lists with dictionaries
- Fixed unit test strings so can see characters in github
- Updated unit tests to use theory/inline data
